packages/hebao_v1/contracts/modules/security/GuardianModule.sol

```diff
@@ -112,22 +112,22 @@ contract GuardianModule is SecurityModule
         emit GuardianRemovalCancelled(wallet, guardian);
     }
 
-    function extractMetaTxSigners(
-        address wallet,
-        bytes4  method,
-        bytes   memory
+    function verifySigners(
+        address   wallet,
+        bytes4    method,
+        bytes     memory /*data*/,
+        address[] memory signers
```

Comment:
I think the `signers` parameter is unnecessary, instead, users can pass in a guardianGroupId and we can get all the signers from securityStore. And in this way, we don't need to validate signers.

Reply:
I don't think that's sufficient because there's too many possibilities. Signers can be from different groups, and each group can have different majority rules. And over all the different groups it is decided if there's a majority or not. Sometimes the owner is needed for a majority, sometimes it is disallowed, other times it is allowed but not required.

By passing the signers separately we have the full flexibility per function without any limitations on the actual function parameters (a limitation which we had until now) which I think is an overall benefit. Now there's not need to add function parameters just so we can extract the signers from it.

Reply:
Okay